adapters/handlers/rest/configure_api.go

```diff
@@ -410,13 +346,85 @@ func configureAPI(api *operations.WeaviateAPI) http.Handler {
 		migrator.RecountProperties(ctx)
 	}
 
+	return appState
+}
+
+func configureAPI(api *operations.WeaviateAPI) http.Handler {
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, 60*time.Minute)
+	defer cancel()
+
+	config.ServerVersion = parseVersionFromSwaggerSpec()
+	appState := MakeAppState(ctx, connectorOptionGroup)
+
+	api.ServeError = openapierrors.ServeError
+
+	api.JSONConsumer = runtime.JSONConsumer()
+
+	api.OidcAuth = composer.New(
+		appState.ServerConfig.Config.Authentication,
+		appState.APIKey, appState.OIDC)
+
+	api.Logger = func(msg string, args ...interface{}) {
+		appState.Logger.WithField("action", "restapi_management").Infof(msg, args...)
+	}
+
+	classifier := classification.New(appState.SchemaManager, appState.ClassificationRepo, appState.DB, // the DB is the vectorrepo
+		appState.Authorizer,
+		appState.Logger, appState.Modules)
+
+	setupSchemaHandlers(api, appState.SchemaManager, appState.Metrics, appState.Logger)
+	objectsManager := objects.NewManager(appState.Locks,
+		appState.SchemaManager, appState.ServerConfig, appState.Logger,
+		appState.Authorizer, appState.DB, appState.Modules,
+		objects.NewMetrics(appState.Metrics))
+	setupObjectHandlers(api, objectsManager, appState.ServerConfig.Config, appState.Logger,
+		appState.Modules, appState.Metrics)
+	setupObjectBatchHandlers(api, appState.BatchManager, appState.Metrics, appState.Logger)
+	setupGraphQLHandlers(api, appState, appState.SchemaManager, appState.ServerConfig.Config.DisableGraphQL,
+		appState.Metrics, appState.Logger)
+	setupMiscHandlers(api, appState.ServerConfig, appState.SchemaManager, appState.Modules,
+		appState.Metrics, appState.Logger)
+	setupClassificationHandlers(api, classifier, appState.Metrics, appState.Logger)
+	backupScheduler := backup.NewScheduler(
+		appState.Authorizer,
+		clients.NewClusterBackups(appState.ClusterHttpClient),
+		appState.DB, appState.Modules,
+		appState.Cluster,
+		appState.Logger)
+	setupBackupHandlers(api, backupScheduler, appState.Metrics, appState.Logger)
+	setupNodesHandlers(api, appState.SchemaManager, appState.DB, appState)
+
+	grpcServer := createGrpcServer(appState)
+	setupMiddlewares := makeSetupMiddlewares(appState)
+	setupGlobalMiddleware := makeSetupGlobalMiddleware(appState)
+
+	api.ServerShutdown = func() {
+		// stop reindexing on server shutdown
+		appState.ReindexCtxCancel()
+
+		// gracefully stop gRPC server
+		grpcServer.GracefulStop()
+
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+
+		if err := appState.SchemaManager.Shutdown(ctx); err != nil {
+			panic(err)
```

Comment:
Perhaps we should just print an error message instead of a panic, and attempt to stop db (line 416), even when schema manager encounters and issue.

Reply:
This isn't actually my code, it's the same code as before, just moved around.

I don't want to start changing it because I'm not sure of the effects, and if a problem occurs, we won't know if it was caused by moving the code or updating it.